src/js/components/BreadcrumbComponent.jsx

```diff
@@ -0,0 +1,87 @@
+var React = require("react/addons");
+var Link = require("react-router").Link;
+
+var ViewHelper = require("../helpers/ViewHelper");
+
+var BreadcrumbComponent = React.createClass({
+  displayName: "BreadcrumbComponent",
+
+  propTypes: {
+    app: React.PropTypes.string,
+    group: React.PropTypes.string,
+    task: React.PropTypes.string
+  },
+
+  getGroupLinks: function () {
+    var group = this.props.group;
+    if (group == null) {
+      return null;
+    }
+
+    var pathParts = group.split("/").slice(0, -1);
```

Comment:
@pierlo-upitup The skip().take() shouldn't be a problem, because of the laziness. The docs suggests this actually. I thought the evaluation of this could be the problem, like @philipnrmn also said, but replaced with a usual slice doesn't speed it up.
My guess is that the native .slice() used beforehands and afterwards the .map is just very very fast and optimized in the js-engine.